server/executor/poller_executor.go

```diff
@@ -159,7 +159,14 @@ func (pe DefaultPollerExecutor) donePollingTraces(job *PollingRequest, traceDB t
 		return false
 	}
 
-	if len(trace.Flat) > traceDB.MinSpanCount() && len(trace.Flat) == len(job.run.Trace.Flat) {
+	hasCollectedTriggeredSpans := len(trace.Flat) > traceDB.MinSpanCount()
+	haveNotCollectedSpansSinceLastPoll := len(trace.Flat) == len(job.run.Trace.Flat)
+
+	// Today we consider that we finished collecting traces
+	// when we collected at least one more span beyond "Tracetest trigger"
+	// and if we haven't collected any new spans since our last poll
+
+	if hasCollectedTriggeredSpans && haveNotCollectedSpansSinceLastPoll {
```

Comment:
```suggestion
	hasCollectedTriggerSpans := len(trace.Flat) > traceDB.MinSpanCount()
	haveNotCollectedSpansSinceLastPoll := len(trace.Flat) == len(job.run.Trace.Flat)

	// Today we consider that we finished collecting traces
	// when we collected at least one more span beyond "Tracetest trigger"
	// and if we haven't collected any new spans since our last poll

	if hasCollectedTriggerSpans && haveNotCollectedSpansSinceLastPoll {
```

I think it's easier to understand "trigger spans" rather than "triggered spans" which might reference the spans that were triggered by the test (i.e., all trace spans)